src/components/plots/FrequencyPlot.jsx

```diff
@@ -0,0 +1,262 @@
+          titleFontSize: { value: config.label.titleFontSize },
+          titleColor: { value: config.colour.masterColour },
+          labelFontSize: { value: config.label.labelFontSize },
+          domainWidth: { value: config.axes.domainWidth },
+        },
+        {
+          orient: 'left',
+          scale: 'y',
+          zindex: 1,
+          gridColor: { value: config.colour.masterColour },
+          gridOpacity: { value: (config.axes.gridOpacity / 20) },
+          gridWidth: { value: (config.axes.gridWidth / 20) },
+          tickColor: { value: config.colour.masterColour },
+          offset: { value: config.axes.offset },
+          title: { value: config.axes.yAxisText },
+          titleFont: { value: config.fontStyle.font },
+          labelFont: { value: config.fontStyle.font },
+          labelColor: { value: config.colour.masterColour },
+          titleFontSize: { value: config.label.titleFontSize },
+          titleColor: { value: config.colour.masterColour },
+          labelFontSize: { value: config.axes.labelFontSize },
+          domainWidth: { value: config.axes.domainWidth },
+        },
+      ],
+
+      marks: [
+        {
+          type: 'rect',
+          from: { data: 'data' },
+          encode: {
+            enter: {
+              x: { scale: 'x', field: 'x' },
+              width: { scale: 'x', band: 1, offset: -1 },
+              y: { scale: 'y', field: 'y0' },
+              y2: { scale: 'y', field: 'y1' },
+              fill: { scale: 'color', field: 'c' },
+            },
+            update: {
+              fillOpacity: { value: 1 },
+            },
+          },
+        },
+      ],
+      legends: legend,
+      title:
+      {
+        text: { value: config.title.text },
+        color: { value: config.colour.masterColour },
+        anchor: { value: config.title.anchor },
+        font: { value: config.fontStyle.font },
+        dx: 10,
+        fontSize: { value: config.title.fontSize },
+      },
+    };
+  };
+
+  const calculateSum = (proportionClusters, xAxisClustersIds) => {
+    let sum = 0;
+
+    if (!xAxisClustersIds.length) {
+      proportionClusters.forEach((cellSetCluster) => {
+        sum += properties[cellSetCluster.key].cellIds.size;
+      });
+      return sum;
+    }
+    proportionClusters.forEach((cellSetCluster) => {
+      const cellSetIds = Array.from(properties[cellSetCluster.key].cellIds);
+      sum += xAxisClustersIds.filter((id) => cellSetIds.includes(id)).length;
+    });
+    return sum;
+  };
+
+  const getClustersInXAxis = (name) => hierarchy.filter((cluster) => (
+    cluster.key === name))[0]?.children;
+
+  const generateData = (spec) => {
+    let data = [];
+
+    const proportionClusters = hierarchy.filter((cluster) => (
+      cluster.key === config.proportionGrouping))[0]?.children;
+
+    if (!proportionClusters) {
+      return [];
+    }
+
+    const clustersInXAxis = getClustersInXAxis(config.xAxisGrouping);
```

Comment:
are there any unit tests for this plot? If not, we should add some, especially for this `getClustersInXAxis` function, which is full of logic